### ISSUE TITLE
wifiMulti : Guru Meditation Error: Core  0 panic'ed (Unhandled debug exception)

### ISSUE THREAD

**Issue by Speedy2k** [espressif/arduino-esp32#3481]
Here is the problem, i have a Guru Meditation error as soon as i run the 
wifiMulti.run()

In my program. I don't understand why.

Here is my current program:

``
#include <Arduino.h>
#include <WiFi.h>
#include <WiFiMulti.h>

TaskHandle_t core0Task;
WiFiMulti wifiMulti;

void setup() {
  enableCore0WDT();
  enableCore1WDT();
  Serial.begin(115200);
  WiFi.mode(WIFI_AP_STA);
  WiFi.setHostname(serial);
  WiFi.softAP(serial);
  IPAddress myIP = WiFi.softAPIP();
  Serial.print("AP IP address: ");
  Serial.println(myIP);
  wifiMulti.addAP("ErabliTEK_SHOP", "0987654321");
  xTaskCreatePinnedToCore(core0Loop, "WiFi&MQTT", 1000, NULL, 5, &core0Task, 0);

unsigned long core0Count;
unsigned long core0Reset;
void core0Loop( void * pvParameters ){
  Serial.print("0: ");Serial.println(millis());
  Serial.print("WiFi running on core: ");
  Serial.println(xPortGetCoreID());
  //esp_int_wdt_cpu_init();
  //esp_task_wdt_add(NULL);
  for(;;){
    //esp_task_wdt_reset();
    Serial.print("1: ");Serial.println(millis());
    core0Count += 1;
    if((millis() - core0Reset) >= 1000){
      core0Hz.add(core0Count);
      core0Count = 0;
      core0Reset = millis();
    }
    wd0Millis = millis();
    Serial.print("2: ");Serial.println(millis());
    if(wifiMulti.run() == WL_CONNECTED){
      Serial.print("3: ");Serial.println(millis());
      if(WiFiConnected == false){
        Serial.print("4: ");Serial.println(millis());
        //init time
        configTzTime(TZ_INFO, ntpServer);
        timeInit = true;
        WiFiConnected = true;
        Serial.println(F("WiFi Connected... yeey :)"));
        Serial.print(F("localIp: "));
        Serial.println(WiFi.localIP());
        getRemoteIp();
        tgetRemoteIp.attach(300, getRemoteIp);
        getWeather();
        tgetWeather.attach(600, getWeather);
        tSendToCloud.attach_ms(1000, sendToCloud);
      }
      Serial.print("5: ");Serial.println(millis());
      if(client.connected()){
        Serial.print("6: ");Serial.println(millis());
        client.loop();
      }else{
        Serial.print("7: ");Serial.println(millis());
        queueInit = false;
        reconnect();
      }
      Serial.print("8: ");Serial.println(millis());
    }else{
      Serial.print("9: ");Serial.println(millis());
      if(WiFiConnected == true){
        Serial.print("10: ");Serial.println(millis());
        remoteIp[0] = '\0';
        WiFiConnected = false;
        Serial.println(F("WiFi Disconnected... oups :("));
        //wm.autoConnect(serial);
        tgetWeather.detach();
        tgetRemoteIp.detach();
        tSendToCloud.detach();
      }
      Serial.print("11: ");Serial.println(millis());
    }
    Serial.print("12: ");Serial.println(millis());
    vTaskDelay(1);
  }
}
``

And it nevers goes pass  if(wifiMulti.run() == WL_CONNECTED){

I got this debug message:

Guru Meditation Error: Core  0 panic'ed (Unhandled debug exception)
Debug exception reason: Stack canary watchpoint triggered (WiFi&MQTT) 
Core 0 register dump:
PC      : 0x4008bb48  PS      : 0x00060c36  A0      : 0x40084f4c  A1      : 0x3ffd89e0  
A2      : 0x00000000  A3      : 0x00060c23  A4      : 0x00060c20  A5      : 0x0000002c  
A6      : 0x00000000  A7      : 0x00000000  A8      : 0x8008578e  A9      : 0x3ffd8b70  
A10     : 0x0000002c  A11     : 0x40085748  A12     : 0x00000000  A13     : 0x00000001  
A14     : 0x00060820  A15     : 0x00000000  SAR     : 0x0000000a  EXCCAUSE: 0x00000001  
EXCVADDR: 0x00000000  LBEG    : 0x400014fd  LEND    : 0x4000150d  LCOUNT  : 0xfffffffe  

Backtrace: 0x4008bb48:0x3ffd89e0 0x40084f49:0x3ffd8ab0 0x4008c267:0x3ffd8ad0 0x4008c8ec:0x3ffd8af0 0x40085254:0x3ffd8b10 0x40085289:0x3ffd8b30 0x4008574d:0x3ffd8b50 0x4000beaf:0x3ffd8b70 0x4008578b:0x3ffd8b90 0x4000bef5:0x3ffd8bb0 0x40081e76:0x3ffd8bd0 0x40081e9d:0x3ffd8bf0 0x4012d29c:0x3ffd8c10 0x400de961:0x3ffd8c40 0x400ddc1d:0x3ffd8c80 0x400d7e3a:0x3ffd8d20 0x400893bd:0x3ffd8d50

That give me this Exception in the exception decoder!

PC: 0x4008bb48
EXCVADDR: 0x00000000

Decoding stack results
0x4008c267: multi_heap_internal_unlock at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/heap/multi_heap.c line 380
0x4008c8ec: multi_heap_malloc at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/heap/multi_heap_poisoning.c line 202
0x40085254: heap_caps_malloc at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/heap/heap_caps.c line 111
0x40085289: heap_caps_malloc_default at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/heap/heap_caps.c line 140
0x4008574d: _malloc_r at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/newlib/syscalls.c line 37
0x4008578b: _calloc_r at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/newlib/syscalls.c line 58
0x40081e76: wifi_calloc at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/esp_adapter.c line 94
0x40081e9d: wifi_zalloc_wrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/esp_adapter.c line 100
0x400de961: WiFiScanClass::scanNetworks(bool, bool, bool, unsigned int) at C:\Users\i7\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4\libraries\WiFi\src\WiFiScan.cpp line 83
0x400ddc1d: WiFiMulti::run(unsigned int) at C:\Users\i7\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4\libraries\WiFi\src\WiFiMulti.cpp line 103
0x400d7e3a: core0Loop(void*) at C:\Users\i7\Documents\GitHub\0403/0403.ino line 1888
0x400893bd: vPortTaskWrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/port.c line 143

Is there anything i do wrong?

**Comment by lbernstone** [comment]
1) If you are going to open a new task to handle WiFi, you should do all your WiFi activities in that task.  I don't recommend this, but otherwise you will run into further problems.
2) Your task uses a very small stack space (1000 bytes).  Start with 4k and work down from there if you are sure your task will not need much stack memory.
